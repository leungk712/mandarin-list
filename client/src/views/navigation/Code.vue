<template>
  <application-layout>
    <v-container class="container-border">
      <h1>Code Overview</h1>
      <v-divider />
      <div class="mt-4">
        <h3>Project Timeline</h3>
        <v-timeline>
          <v-timeline-item color="red lighten-2"
            >Jan '21 - Project idea enters brain</v-timeline-item
          >
          <v-timeline-item class="text-right" color="purple lighten-2">
            Jan '21 - Mar '21 - Coding & suffering
          </v-timeline-item>
          <v-timeline-item color="green lighten-2"
            >Mar '21 - Project v1.0 release</v-timeline-item
          >
          <v-timeline-item class="text-right"
            >Apr '21 - Beyond - ???</v-timeline-item
          >
        </v-timeline>
      </div>
      <v-divider class="mt-4" />
      <div class="mt-4">
        <v-row>
          <v-col cols="6">
            {{ codeNote }}
            <v-row class="mt-4 ml-1">
              <div class="mr-4">
                <a
                  href="https://github.com/leungk712"
                  target="_blank"
                  class="code-icon github-icon"
                  title="Check out my Github account!"
                >
                  <i class="fab fa-github fa-2x"></i>
                </a>
              </div>
              <div>
                <a
                  href="https://www.linkedin.com/in/leungk712/"
                  target="_blank"
                  class="code-icon linkedin-icon"
                  title="Check out my LinkedIn account!"
                >
                  <i class="fab fa-linkedin fa-2x"></i>
                </a>
              </div>
            </v-row>
          </v-col>
          <v-col cols="6">
            <h3>Project Resources</h3>
            <v-divider class="my-2" />
            <ul v-for="(resource, idx) in projectResources" :key="idx">
              <li>
                <a
                  class="text-decoration-none"
                  :href="resource.url"
                  target="_blank"
                >
                  {{ resource.name }}
                </a>
              </li>
            </ul>
          </v-col>
        </v-row>
      </div>
    </v-container>
  </application-layout>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import ApplicationLayout from "@/components/layouts/ApplicationLayout.vue";
import { codeNote, projectResources } from "@/helpers/code.ts";

@Component({
  name: "Code",
  components: { ApplicationLayout }
})
export default class Code extends Vue {
  // ===== Store ===== //

  // ===== Data ===== //
  public projectResources = projectResources;
  public codeNote = codeNote;

  // ===== Methods ===== //

  // ===== Computed ===== //

  // ===== Lifecycle Hooks ===== //
}
</script>

<style scoped>
/* .fa-github, .fa-linkedin {
    color: black;
  } */

.github-icon {
  color: #6e5494;
}

.linkedin-icon {
  color: #0072b1;
}
</style>
