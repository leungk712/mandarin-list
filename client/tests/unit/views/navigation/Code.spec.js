import { createLocalVue, shallowMount } from "@vue/test-utils";
import Code from "@/views/navigation/Code.vue";
import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";

Vue.use(Vuetify);

const localVue = createLocalVue();
localVue.use(Vuex);

const elem = document.createElement("div");
if (document.body) {
    document.body.appendChild(elem);
};

document.body.setAttribute("data-app", true);

const actions = {};

const mockStore = {};

describe('Code', () => {
    let wrapper;
    const store = new Vuex.Store({
        modules: mockStore
    });
    const mountFunction = (options, store = mockStore) => {
        return shallowMount(Code, {
           attachTo: elem,
           localVue,
           store,
           vuetify: new Vuetify(),
           ...options
        });
    };
    describe("template", () => {
        it("should match snapshot", () => {
            wrapper = mountFunction({}, store);
            expect(wrapper.vm.$el).toMatchSnapshot();
            wrapper.destroy();
        });
    });
});