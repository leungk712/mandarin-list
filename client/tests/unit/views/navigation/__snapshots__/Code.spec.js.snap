// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code template should match snapshot 1`] = `
<application-layout-stub>
  <v-container-stub
    class="container-border code-container"
    tag="div"
  >
    <h1>
      Code Overview
    </h1>
     
    <v-divider-stub />
     
    <div
      class="mt-4"
    >
      <h3>
        Project Timeline
      </h3>
       
      <v-timeline-stub>
        <v-timeline-item-stub
          color="red lighten-2"
        >
          Jan '21 - Project idea enters brain
        </v-timeline-item-stub>
         
        <v-timeline-item-stub
          class="text-right"
          color="purple lighten-2"
        >
          
          Jan '21 - Mar '21 - Coding & suffering
        
        </v-timeline-item-stub>
         
        <v-timeline-item-stub
          color="green lighten-2"
        >
          Mar '21 - Project v1.0 release
        </v-timeline-item-stub>
         
        <v-timeline-item-stub
          class="text-right"
          color="primary"
        >
          Apr '21 - Beyond - ???
        </v-timeline-item-stub>
      </v-timeline-stub>
    </div>
     
    <v-divider-stub
      class="mt-4"
    />
     
    <div
      class="mt-4"
    >
      <v-row-stub
        tag="div"
      >
        <v-col-stub
          cols="6"
          tag="div"
        >
          
          This project is by far the most favorite project I've ever created. This project wouldn't be possible without very smart
    people who have come before me paving the way so that I can create this project with the knowledge and resources to execute my vision. 
    As I continue to create new projects that excite me, I want to continue to prioritize building this project out with future bug fixes, new features, and
    continued improvement of the user interface and experience. Thank you for using this application. - KL
          
          <v-row-stub
            class="mt-4 ml-1"
            tag="div"
          >
            <div
              class="mr-4"
            >
              <a
                class="code-icon github-icon"
                href="https://github.com/leungk712"
                target="_blank"
                title="Check out my Github account!"
              >
                <i
                  class="fab fa-github fa-2x"
                />
              </a>
            </div>
             
            <div>
              <a
                class="code-icon linkedin-icon"
                href="https://www.linkedin.com/in/leungk712/"
                target="_blank"
                title="Check out my LinkedIn account!"
              >
                <i
                  class="fab fa-linkedin fa-2x"
                />
              </a>
            </div>
          </v-row-stub>
           
          <v-divider-stub
            class="mt-6"
          />
           
          <div
            class="mt-6"
          >
            <p>
              Below is the image of my first iteration of this project back in January. No user profiles, categories, stories.
            </p>
             
            <v-img-stub
              class="mt-6"
              options="[object Object]"
              position="center center"
              src="@/assets/code/mandarin-list-original.png"
              transition="fade-transition"
            />
          </div>
        </v-col-stub>
         
        <v-col-stub
          cols="6"
          tag="div"
        >
          <h3>
            Project Resources
          </h3>
           
          <v-divider-stub
            class="my-2"
          />
           
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://vuejs.org/"
                target="_blank"
              >
                
                Vue.js
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://vuex.vuejs.org/"
                target="_blank"
              >
                
                Vuex
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://router.vuejs.org/"
                target="_blank"
              >
                
                VueRouter
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://vuetifyjs.com/en/"
                target="_blank"
              >
                
                Vuetify
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://vee-validate.logaretm.com/v4/"
                target="_blank"
              >
                
                VeeValidate
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://momentjs.com/"
                target="_blank"
              >
                
                moment.js
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://material.io/icons/"
                target="_blank"
              >
                
                Material Icon
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://materialdesignicons.com/"
                target="_blank"
              >
                
                Material Design Icons
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://fontawesome.com/"
                target="_blank"
              >
                
                Font Awesome
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://unsplash.com/"
                target="_blank"
              >
                
                Unsplash (beautiful photos)
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://www.npmjs.com/package/number-to-chinese-words"
                target="_blank"
              >
                
                Number-to-Chinese-Words
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://www.npmjs.com/package/chinese-to-pinyin"
                target="_blank"
              >
                
                Chinese-to-Pinyin
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://github.com/axios/axios"
                target="_blank"
              >
                
                Axios
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://nodejs.org/en/"
                target="_blank"
              >
                
                Node.js
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://expressjs.com/"
                target="_blank"
              >
                
                Express
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://mongoosejs.com"
                target="_blank"
              >
                
                Mongoose
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://www.mongodb.com/"
                target="_blank"
              >
                
                MongoDB
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://www.heroku.com/home"
                target="_blank"
              >
                
                Heroku
              
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="https://www.netlify.com/"
                target="_blank"
              >
                
                Netlify
              
              </a>
            </li>
          </ul>
        </v-col-stub>
      </v-row-stub>
    </div>
  </v-container-stub>
</application-layout-stub>
`;
